src/components/ModelProvider.tsx

```diff
@@ -88,6 +89,11 @@ export class ModelProvider extends Component<ModelProviderType> {
               const props = Utils.modelToProps(data);
 
               this.setState(props);
+
+              // Fire event once component model has been fetched and rendered to enable editing on AEM
+              if (injectPropsOnInit) {
+                  PathUtils.dispatchGlobalCustomEvent(Constants.REMOTE_CONTENT_LOADED, {});
```

Comment:
This event needs to be fired in WCM_Edit mode. Maybe, we should restrict it to that.
Also, as discussed, maybe, we can add a similar check for `cq-pagemodel-loaded` event as well. Wdyt ?

cc @pfauchere

Reply:
I've updated it to fire only within the editor. However, the `cq-pagemodel-loaded` event has a wider scope and is used in multiple areas. We could tackle it as a different issue possibly. WDYT @habansal?

Reply:
I am fine with that. Thanks!